spi/common/core-spi/src/main/java/org/eclipse/edc/spi/system/health/HealthCheckResult.java

```diff
@@ -43,6 +44,11 @@ public static HealthCheckResult failed(String... errors) {
         return new HealthCheckResult(false, new Failure(errorList));
     }
 
+    public static HealthCheckResult failed(List<String> errors) {
```

Comment:
this seems fishy: why can this list ever contain `null` elements?

Reply:
You have no control of what downstream projects will call this method with so better check for null

Reply:
looks too defensive for me as well, a null error message is just a null error message, won't hurt anyone